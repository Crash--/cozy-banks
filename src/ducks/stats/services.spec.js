import {
  getPeriod,
  fetchTransactionsForPeriod,
  getMeanOnPeriod
} from './services'
import MockDate from 'mockdate'
import { BankTransaction } from 'cozy-doctypes'
import { endOfDay } from 'date-fns'

jest.mock('cozy-doctypes')

describe('getPeriod', () => {
  afterEach(() => {
    MockDate.reset()
  })

  it('should return the period', () => {
    MockDate.set(new Date('2019-07-08T00:00:00.000Z'))

    const period = getPeriod()

    expect(period).toMatchSnapshot()
  })
})

describe('fetchTransactionsForPeriod', () => {
  it('should fetch transactions for the given period', async () => {
    const period = {
      start: new Date('2019-03-01'),
      end: endOfDay(new Date('2019-06-30'))
    }

    await fetchTransactionsForPeriod(period)

    expect(BankTransaction.queryAll).toHaveBeenCalledWith({
      date: {
        $gte: period.start,
        $lte: period.end
      }
    })
  })
})

describe('getMeanOnPeriod', () => {
  it('should return the mean amount by month on the given period', () => {
    const transactions = [{ amount: 100 }, { amount: 50 }]
    const period = {
      start: new Date('2019-03-01'),
      end: endOfDay(new Date('2019-06-30'))
    }

    expect(getMeanOnPeriod(transactions, period)).toBe(50)
  })

  it('should return a positive value even if transactions amount is negative', () => {
    const transactions = [{ amount: -100 }, { amount: -50 }]
    const period = {
      start: new Date('2019-03-01'),
      end: endOfDay(new Date('2019-06-30'))
    }

    expect(getMeanOnPeriod(transactions, period)).toBe(50)
  })
})
