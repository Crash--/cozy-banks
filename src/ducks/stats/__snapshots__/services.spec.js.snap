// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getPeriod should return the period 1`] = `
Object {
  "end": "2019-06-30T23:59:59.999Z",
  "start": "2019-04-01T00:00:00.000Z",
}
`;
